tools/codegen/pkg/generation/generator.go

```diff
@@ -6,7 +6,7 @@ type Generator interface {
 	Name() string
 
 	// GenGroup runs the generator against the given APIGroupContext.
-	GenGroup(APIGroupContext) error
+	GenGroup(APIGroupContext) ([]Result, error)
```

Comment:
Something to note here, if any of the generators copy errors into both the results that are returned and aggregate them into a single error to return, the output will contain both the new result based formatting _and_ a blob of red error text.

For example, the output in https://prow.ci.openshift.org/view/gs/test-platform-results/pr-logs/pull/openshift_api/2093/pull-ci-openshift-api-master-verify-crd-schema/1856782506544599040#1:build-log.txt%3A540-549 contains the same error message twice. In this particular case it is likely because the schemachecker is returning the same error in both return values.

This raises the question for me - is this what we want? 
Generally, I would recommend we have a single output format that we update all the generators to use so that it isn't confusing to those looking at the output which set of information to look at (and then after some time realize they contain the same error information).

Would it be beneficial to make _all_ generators return only a slice of `[]Result` and if we notice the set of results from all generators contains errors we have a single error that says something like "generators [....] returned errors, see the report for more information" ?

Reply:
Had a separate chat with @JoelSpeed in Slack and he mentioned this being an intermediate state. I think this is fine as is for an intermediate state, but when deciding on the end state formatting we should ensure that there is consistency that reduces room for confusion of those looking into failures related to these generators.